src/api/search_api.rs

```diff
@@ -2,24 +2,32 @@ use crate::common::helpers::process_response;
 use actix_web::rt::time::Instant;
 use actix_web::{post, web, Responder};
 use collection::operations::types::SearchRequest;
+use segment::types::ScoredPoint;
 use std::sync::Arc;
+use storage::content_manager::errors::StorageError;
 use storage::content_manager::toc::TableOfContent;
 
+async fn do_search_points(
+    toc: &TableOfContent,
+    name: &str,
+    request: SearchRequest,
+) -> Result<Vec<ScoredPoint>, StorageError> {
+    toc.get_collection(&name)?
+        .search(Arc::new(request))
+        .await
+        .map_err(|err| err.into())
+}
+
 #[post("/collections/{name}/points/search")]
 pub async fn search_points(
     toc: web::Data<TableOfContent>,
-    web::Path(name): web::Path<String>,
```

Comment:
In a new actix you can't do ``web::Path(name)`` anymore, you have to use ``into_inner``.